### PR TITLE
Issue #1581 Add GitHub issue template

### LINKED ISSUE #1581: Add .github/ISSUE_TEMPLATE.md
### General information

  * Minishift version: 
  * Download from: github releases / brew / chocolatey
  * OS: Linux / macOS / Windows
  * OS version/distro: 
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox
  * `minishift start --show-libmachine-logs -v5`
    If long, please consider posting this on http://gists.github.com/ and post the link in your issue

#### Steps to reproduce

  1. 
  2. 
  3. 
  4.

### PR DESCRIPTION
**Pull request by gbraad**
Fixes #1581

Opened PR for feedback...

### CODE REVIEW

**Comment by budhrg** [comment]
@gbraad  See https://github.com/minishift/minishift/pull/239 which I created long back :smile:

**Comment by gbraad** [comment]
I think this is valid to discuss again... I dislike getting single-line issue reports or having to ask for missing logs.

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+### General information
+
+  * Minishift version: 
+  * Download from: github releases / brew / chocolatey
+  * OS: Linux / macOS / Windows
+  * OS version/distro: 
+  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox
+  * `minishift start --shwo-libmachine-logs -v5`
```
Comment (robin-owen) [inline .github/ISSUE_TEMPLATE.md]: Typo: `--show-libmachine-logs`

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    If long, please consider posting this on http://gists.github.com/ and post the link in your issue
```
Comment (robin-owen) [inline .github/ISSUE_TEMPLATE.md]: The link should be https://gist.github.com/ -- gists.github.com results in a 404.

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (gbraad) [reply .github/ISSUE_TEMPLATE.md]: This one was noted after commit ;-) https://github.com/gbraad/minishift/commit/a636f6d078ae6e3c1c6635f32390b010b32e9aeb#commitcomment-25034542. The idea to publish was to get team feedback... so far, not much.

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (gbraad) [reply .github/ISSUE_TEMPLATE.md]: a paste service, like pastebin.centos.org might also be a solution to mention

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (robin-owen) [reply .github/ISSUE_TEMPLATE.md]: Ha, indeed it was -- missed that.

Personally, though, I think having an issue template is a good idea.

**Comment by hferentschik** [comment]
I am not a fan of it. I already made my case in #239. IMO it gets in the way more often than not. Different in systems like JIRA and co. But Iet the majority of @minishift/minishift-dev decide. -1 from me.

**Comment by gbraad** [comment]
I have seen several issues in recent days, with a oneliner and a log that explain nothing of the real issue they are facing. Compare that with:

https://github.com/minishift/minishift/issues/1615
https://github.com/minishift/minishift/issues/1520

Please can still choose to CTRL+A and delete the text, so I do not see what the actual problem is... It gives people a guide, and maybe even the solution, for the problem they are facing. As often, just reading the log output will explain what went wrong.

**Comment by anjannath** [comment]
Well the idea of an issue template is great, it'll structure the information and we can understand the problem better, but i don't think we need the actual and expected output, as the logs are already attached.

**Comment by gbraad** [comment]
@anjannath expected and actual are needed in case the implementation/result is different. This can happen in case when OpenShift gives an output, but our wrapping does not work as expected, or when `oc cluster up` might not support something correctly. The template can help people to structure the formulation...

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+### General information
+
+  * Minishift version: 
+  * Download from: github releases / brew / chocolatey
+  * OS: Linux / macOS / Windows
+  * OS version/distro: 
+  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox
```
Comment (LalatenduMohanty) [inline .github/ISSUE_TEMPLATE.md]: If we are asking for output of `minishift start --shwo-libmachine-logs -v5` then we do not need to ask which hypervisor user is using. Also I am not sure if we get the OS information, Minishift Version etc in the output. If not then we should add these to the debug logs. basically we should ask very minimal things from user.

**Comment by LalatenduMohanty** [comment]
I strongly feel we we should have an issue template. Because sometimes users do not know what to provide which will help developers and there is no point going back and forth for information. 
However I also strongly feel that the template should be minimal in nature. We should just ask following information
```
### Problem description

#### Expected

#### Actual

#### Steps to reproduce

#### Out put of `minishift start --shwo-libmachine-logs -v5` if it is a bug. 

```
I would suggest we should first make sure `minishift start --shwo-libmachine-logs -v5` has all required information (as I mentioned in https://github.com/minishift/minishift/pull/1582#discussion_r147324538) and then merge the template.

**Comment by LalatenduMohanty** [comment]
Also template is subject to change in future. Depending upon our experience we will add or remove fields. I would suggest we start from a minimal template and then add or remove in future.

**Comment by agajdosi** [comment]
@LalatenduMohanty Adding just a small thought I got - maybe it could be nice to have some kind of alias for `--show-libmachine-logs -v 5` as it is still not really easy to remember. For example we could have `--debug` or `--report` which would take care of libmachine logs and -v 5 and would make it more friendly for user to report. Do you think it could be useful? I can create a suggestion issue, if this idea is not completely off.

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+### General information
+
+  * Minishift version: 
+  * Download from: github releases / brew / chocolatey
```
Comment (budhrg) [inline .github/ISSUE_TEMPLATE.md]: This line is unnecessary IMO.

File: .github/ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  * OS: Linux / macOS / Windows
+  * OS version/distro:
```
Comment (budhrg) [inline .github/ISSUE_TEMPLATE.md]: The lines 6 to 7 are unnecessary IMO. Agree with @LalatenduMohanty  that `--show-libmachine-logs -v 5` will display which hypervisor user is using.

File: .github/ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  * Minishift version: 
+  * Download from: github releases / brew / chocolatey
+  * OS: Linux / macOS / Windows
+  * OS version/distro: 
+  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox
+  * `minishift start --shwo-libmachine-logs -v5`
+    If long, please consider posting this on http://gists.github.com/ and post the link in your issue
+
+### Problem description
```
Comment (budhrg) [inline .github/ISSUE_TEMPLATE.md]: I think title will include part of description so this is unnecessary IMO too for issue template context.

File: .github/ISSUE_TEMPLATE.md (hunk 5 above)
Comment (LalatenduMohanty) [reply .github/ISSUE_TEMPLATE.md]: @budhrg I guess there will be some issues where user do not need to run `--show-libmachine-logs -v5 `. So if user is giving out put of `--show-libmachine-logs -v5 ` then, user does not need to give these details.

**Comment by praveenkumar** [comment]
@gbraad looks like everyone put +1 here except @hferentschik I think you should make changes which requested for this PR and we are good to merge it.

**Comment by budhrg** [comment]
@minishift/minishift-dev  Following is my minimal issue template content I would like to have:

```
### General information

  * Minishift version: 
  * OS: Linux / macOS / Windows
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox

#### Expected

#### Actual

#### Steps to reproduce

  1. 
  2. 

#### Debug logs [Optional]
`minishift start --shwo-libmachine-logs -v 5`
If long, please consider posting this on http://gists.github.com/ and post the link here.

```

**Comment by gbraad** [comment]
I will make needed changes and repropose. Some of the feedback I agree with, and others can be modified to fit.

**Comment by budhrg** [comment]
> I will make needed changes and repropose. Some of the feedback I agree with, and others can be modified to fit.

@gbraad Let's sort this out today. It is hanging here for quite a while.

**Comment by budhrg** [comment]
@LalatenduMohanty @praveenkumar @gbraad @anjannath , this is the minimal issue template I could think of https://github.com/minishift/minishift/pull/1582#issuecomment-344248081.

```
### General information

  * Minishift version: 
  * OS: Linux / macOS / Windows
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox

#### Expected

#### Actual

#### Steps to reproduce

  1. 
  2. 

#### Debug logs [If Applicable]
`minishift start --shwo-libmachine-logs -v 5`
If long, please consider posting this on http://gists.github.com/ and post the link here.
```

Let's get this done.

**Comment by gbraad** [comment]
@minishift/minishift-dev Updated

File: .github/ISSUE_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+### Expected
+
+
+### Actual
+
+
+### Logs
+
+You can start Minishift with `minishift start --show-libmachine-logs -v5` to collect logs.
+Please consider posting this on http://gist.github.com/ and post the link in the issue
```
Comment (budhrg) [inline .github/ISSUE_TEMPLATE.md]: Missing full stop here.

**Comment by LalatenduMohanty** [comment]
> @LalatenduMohanty Adding just a small thought I got - maybe it could be nice to have some kind of alias for --show-libmachine-logs -v 5 as it is still not really easy to remember. For example we could have --debug or --report which would take care of libmachine logs and -v 5 and would make it more friendly for user to report. Do you think it could be useful? I can create a suggestion issue, if this idea is not completely off.

@agajdosi I like `--debug`, but we need to keep things like this same as `minikube`. So the idea is worth pushing in minikube.

**Pull request merged**